File: CommerceFlow/Orders/Shipment/Shipment.cs
namespace CommerceFlow.Orders;
public record Shipment
{
    public ShipmentStatus Status { get; private set; }
    public string? TrackingCode { get; private set; }
    public ShippingAddress? Address { get; set; }

    public void Request()
    {
        Status = ShipmentStatus.Requested;
    }

    public void Dispatch(string trackingCode)
    {
        TrackingCode = trackingCode;
        Status = ShipmentStatus.Dispatched;
    }

    public void Deliver()
    {
        Status = ShipmentStatus.Delivered;
    }

    public static Shipment Create(ShippingAddress shippingAddress)
    {
        var shipment = new Shipment
        {
            Status = ShipmentStatus.Created,
            Address = shippingAddress
        };
        return shipment;
    }
}

File: CommerceFlow.Shipments.WebApp/src/types/shipment.ts
export type ShipmentStatus =
  | "Created"
  | "CarrierAssigned"
  | "Packed"
  | "Dispatched"
  | "Delivered"
  | "Cancelled";

export interface Shipment {
  id: string;
  orderNumber: string;
  createdAt: string;
  status: ShipmentStatus;
  address?: {
    street?: string;
    number?: string;
    complement?: string;
    neighborhood?: string;
    city?: string;
    state?: string;
    zipCode?: string;
  };
  carrier?: { name?: string };
  tracking?: {
    trackingCode?: string;
    events?: Array<{ occurredAt: string; description: string; location: string }>;
  };
}
